### ISSUE TITLE
Bitbucket server pull request state should be case sensitive

### ISSUE THREAD

**Issue by andreami** [fthomas/scala-steward#1100]
Hi guys,

when I run the project with bitbucket-server option I get an error message from circe:

`Caused by: io.circe.DecodingFailure$$anon$2: Unexpected string 'OPEN': DownField(state),DownArray,DownField(values)`

This happens because Bitbucket Server uses uppercase pull request state and the decoder parses lowercase values.

Here is the bitbucket server documentation:
https://docs.atlassian.com/bitbucket-server/rest/6.5.0/bitbucket-rest.html

I resolved the problem with a toLowerCase

```
_.toLowerCase match {
        case "open" | "opened"   => Right(Open)
        case "closed" | "merged" => Right(Closed)
        case unknown             => Left(s"Unexpected string '$unknown'")
      }
```

If you want I can open a Pull Request.

Thanks for the help and congrats!

Andrea

P.S. What if I add "declined" as a Closed status?

**Comment by fthomas** [comment]
Both changes sound good to me and a PR would be welcome!

**Comment by fthomas** [comment]
Fixed by #1102.
